### DISCUSSION TITLE
JDOS calculation

### DISCUSSION THREAD

**Discussion by GH-phys** [wannier-berri/wannier-berri#363] [Q&A]
Hi, I'm trying to calculate joint density of states ,however, error occurs.
My optical conductivity calculation is working fine but JDOS is not.
The error message is 
TypeError: Can't instantiate abstract class JDOS with abstract method factor_omega

Does anyone have idea to solve this problem?
Thank you so much in advance

**Comment by stepan-tsirkin** [comment]
It is a bug. Fixed on PR #364  (will be merged soon)

It should be like this:

https://github.com/wannier-berri/wannier-berri/blob/dad845f8e52cc9496d38fd1203fc9716464872b5/wannierberri/calculators/dynamic.py#L156-L173

**Reply by GH-phys** [reply]
Thanks for the answer, it's working
